arch/risc-v/src/mpfs/Make.defs

```diff
@@ -32,6 +32,7 @@ CMN_CSRCS += riscv_releasestack.c riscv_stackframe.c riscv_schedulesigaction.c
 CMN_CSRCS += riscv_sigdeliver.c riscv_unblocktask.c riscv_usestack.c
 CMN_CSRCS += riscv_mdelay.c riscv_udelay.c riscv_copyfullstate.c
 CMN_CSRCS += riscv_idle.c riscv_tcbinfo.c
+CMN_CSRCS += riscv_mhartid.c
```

Comment:
Consider change to `riscv_cpuindex.c`

Reply:
Yes, I'll remove the duplicate implementation

Reply:
Seems like the declaration of up_cpuindex is dependent on CONFIG_SMP. I suggest I change the name of the function to riscv_cpuindex and 
#define up_cpuindex riscv_cpuindex?

Reply:
Maybe we can decouple up_cpuindex with CONFIG_SMP on RISC-V ?
mhartid is also availabe on non-SMP platform.

Reply:
Yes, I provided a solution / suggestion how to do this. Hartid is needed regardless of CONFIG_SMP.